src/index.js

```diff
@@ -150,11 +140,11 @@ class UglifyJsPlugin {
 
   apply(compiler) {
     const requestShortener = new RequestShortener(compiler.context);
-    // Copy uglify options
-    const uglifyOptions = UglifyJsPlugin.buildDefaultUglifyOptions(this.uglifyOptions);
-    // Making sure output options exists if there is an extractComments options
-    if (this.options.extractComments) {
-      uglifyOptions.output = uglifyOptions.output || {};
+    const { uglifyOptions } = this.options;
+    const uglifiedAssets = new WeakSet();
+
+    if (typeof this.options.sourceMap === 'undefined' && (compiler.options.devtool === 'sourcemap' || compiler.options.devtool === 'source-map')) {
```

Comment:
You're right. I can separate this into a diff pr if u like. 

However, I gave this a good amount of thought and I believe it's been a missing piece to this plugin for some time now.

Not to mention - [this same logic is built into webpack core](https://github.com/webpack/webpack/blob/0ab36c491ea547add35af0c3fbbcdfab03a4c15e/bin/convert-argv.js#L441). I believe this responsibility belongs in this plugin.

If this pr is merged i plan to follow up by making a pr at webpack core repo to extract this logic from there.